dotnetv3/S3/LifecycleExample/LifecycleExample/Lifecycle.cs

```diff
@@ -0,0 +1,202 @@
+                lifeCycleConfiguration = await RetrieveLifecycleConfigAsync(client, bucketName);
+
+                // Add another rule to the existing lifecycle rule.
+                lifeCycleConfiguration.Rules.Add(new LifecycleRule
+                {
+                    Id = "NewRule",
+                    Filter = new LifecycleFilter()
+                    {
+                        LifecycleFilterPredicate = new LifecyclePrefixPredicate()
+                        {
+                            Prefix = "YearlyDocuments/",
+                        },
+                    },
+                    Expiration = new LifecycleRuleExpiration()
+                    {
+                        Days = 3650,
+                    },
+                });
+
+                // Add the lifecycle configuration to the bucket.
+                await AddExampleLifecycleConfigAsync(client, bucketName, lifeCycleConfiguration);
+
+                // Verify that the the bucket now has three rules in its lifecycle configuration.
+                lifeCycleConfiguration = await RetrieveLifecycleConfigAsync(client, bucketName);
+                Console.WriteLine($"Expected # of rulest: 3; # of rules found: {lifeCycleConfiguration.Rules.Count}");
+
+                // Now delete the configuration from the bucket.
+                await RemoveLifecycleConfigAsync(client, bucketName);
+
+                // Show that the lifecycle configuration is no longer associated with
+                // the bucket.
+                lifeCycleConfiguration = await RetrieveLifecycleConfigAsync(client, bucketName);
+            }
+            catch (AmazonS3Exception ex)
+            {
+                Console.WriteLine($"Error encountered ***. Message:'{ex.Message}' when writing an object.");
+            }
+        }
+
+        /// <summary>
+        /// Adds lifecycle configuration information to the S3 bucket named in
+        /// the bucketName parameter.
+        /// </summary>
+        /// <param name="client">The S3 client used to call the
+        /// PutLifecycleConfigurationAsync method.</param>
+        /// <param name="bucketName">A string representing the S3 bucket to
+        /// which configuration information will be added.</param>
+        /// <param name="configuration">A LifecycleConfiguration object that
+        /// will be applied to the S3 bucket.</param>
+        public static async Task AddExampleLifecycleConfigAsync(IAmazonS3 client, string bucketName, LifecycleConfiguration configuration)
+        {
+            PutLifecycleConfigurationRequest request = new ()
+            {
+                BucketName = bucketName,
+                Configuration = configuration,
+            };
+            var response = await client.PutLifecycleConfigurationAsync(request);
+        }
+
+        /// <summary>
+        /// Returns a configuration object for the supplied bucket name.
+        /// </summary>
+        /// <param name="client">The S3 client object used to call
```

Comment:
The S3 -> The Amzon S3

Reply:
Same as before S3 only third reference and after.